matrixio-core.c

```diff
@@ -77,7 +77,8 @@ int matrixio_hw_reg_read(void *context, unsigned int reg, unsigned int *val)
 	hw_addr->burst = 0;
 	hw_addr->readnwrite = 1;
 
-	ret = matrixio_spi_transfer(matrixio->spi, send_buf, (uint8_t *)recv_buf, 4);
+	ret = matrixio_spi_transfer(matrixio->spi, send_buf,
+				    (uint8_t *)recv_buf, 4);
 
 	if (ret < 0) {
 		*val = 0;
```

Comment:
No need to modify val here. The return value should be enough.

Reply:
Below I see *val = recv_buf[1];
Doesn't this cause a memory overflow?

Reply:
it does not cause overflow, (*val), is a value of 32 bits, recv_buf[1] is a value of 16 bits.

Reply:
Yeah. I misread this.